### PR TITLE
Use type safe method to get secret and instanceid

### PR DESCRIPTION
**Pull request by J0WI**
## Summary
Getting rid off some more `getSystemValue()`, see #https://github.com/nextcloud/server/pull/35581 for context.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: core/BackgroundJobs/BackgroundCleanupUpdaterBackupsJob.php [hunk 1]
```diff
@@ -46,9 +46,9 @@ public function __construct(IConfig $config, LoggerInterface $log, ITimeFactory
 	 */
 	public function run($arguments) {
 		$updateDir = $this->config->getSystemValue('updatedirectory', null) ?? $this->config->getSystemValue('datadirectory', \OC::$SERVERROOT . '/data');
-		$instanceId = $this->config->getSystemValue('instanceid', null);
+		$instanceId = $this->config->getSystemValueString('instanceid');
 
-		if (!is_string($instanceId) || empty($instanceId)) {
+		if (empty($instanceId)) {
```
Comment (blizzz) [inline core/BackgroundJobs/BackgroundCleanupUpdaterBackupsJob.php:51]: In `apps/files/lib/Command/ScanAppData.php` the comparison is for an empty string (strictly). Would be good to consolidate. I'd use the strict check.

File: lib/private/Repair/MoveUpdaterStepFile.php [hunk 2]
```diff
@@ -44,9 +44,9 @@ public function getName() {
 
 	public function run(IOutput $output) {
 		$updateDir = $this->config->getSystemValue('updatedirectory', null) ?? $this->config->getSystemValue('datadirectory', \OC::$SERVERROOT . '/data');
-		$instanceId = $this->config->getSystemValue('instanceid', null);
+		$instanceId = $this->config->getSystemValueString('instanceid');
 
-		if (!is_string($instanceId) || empty($instanceId)) {
+		if (empty($instanceId)) {
```
Comment (blizzz) [inline lib/private/Repair/MoveUpdaterStepFile.php:49]: another occurrence

**Review by blizzz — APPROVED** [review]
CI needs https://github.com/nextcloud/server/pull/35580, but there *might* be some tests needing adjustments (didn't check)

File: apps/settings/lib/Mailer/NewUserMailHelper.php [hunk 3]
```diff
@@ -108,7 +108,7 @@ public function generateTemplate(IUser $user, $generatePasswordResetToken = fals
 			);
 			$tokenValue = $this->timeFactory->getTime() . ':' . $token;
 			$mailAddress = (null !== $user->getEMailAddress()) ? $user->getEMailAddress() : '';
```
Comment (come-nc) [inline apps/settings/lib/Mailer/NewUserMailHelper.php:110]:
```suggestion
			$mailAddress = $user->getEMailAddress() ?? '';
```

**Review by come-nc — APPROVED** [review]

**Comment by PVince81** [comment]
@J0WI can you rebase? the bot can't rebase on forks

**Comment by come-nc** [comment]
Failed tests are worrying, cypress is failing and drone nodb has:
```
There was 1 error:
239	
240	1) Test\Security\VerificationToken\VerificationTokenTest::testTokenSuccess
241	OCP\Security\VerificationToken\InvalidTokenException: 
242	
243	/drone/src/lib/private/Security/VerificationToken/VerificationToken.php:71
244	/drone/src/lib/private/Security/VerificationToken/VerificationToken.php:91
245	/drone/src/tests/lib/Security/VerificationToken/VerificationTokenTest.php:290
```

**Comment by szaimen** [comment]
@J0WI nodb failure seems to be related. Also can you please fix the conflicts? Thanks a lot! :)

**Pull request closed without merging**